src/metabase/analyze/fingerprint/insights.clj

```diff
@@ -19,17 +19,19 @@
 (set! *warn-on-reflection* true)
 
 (defn- last-2 []
-  (fn
-    ([] [])
-    ([acc]
-     (let [cnt (count acc)]
-       (cond (= cnt 0) [nil nil]
-             (= cnt 1) [nil (nth acc 0)]
-             :else acc)))
-    ([acc x]
-     (if (< (count acc) 2)
-       (conj acc x)
-       [(nth acc 1) x]))))
+  (let [null (Object.)]
+    (fn
+      ([] (object-array [null null]))
```

Comment:
Is specializing this to a double array not worth it? You could use a volatile to track the "size" instead of "null" objects.

Reply:
Ah good points, and I didn't think about that overhead 🤦